tephra-core/src/main/java/org/apache/tephra/distributed/TransactionServiceThriftClient.java

```diff
@@ -112,7 +113,20 @@ public Transaction startShort() throws TException {
 
   public Transaction startShort(int timeout) throws TException {
     try {
-      return TransactionConverterUtils.unwrap(client.startShortTimeout(timeout));
+      return TransactionConverterUtils.unwrap(client.startShortWithTimeout(timeout));
+    } catch (TGenericException e) {
+      isValid.set(false);
+      // currently, we only expect IllegalArgumentException here, if the timeout is invalid
+      IllegalArgumentException actual;
+      try {
+        actual = (IllegalArgumentException)
+          Class.forName(e.getOriginalExceptionClass())
+            .getConstructor(String.class)
+            .newInstance(e.getMessage());
+      } catch (Throwable t) {
```

Comment:
Because we do not expect anything else than IllegalArgumentException? For now, I will change the code to be more readable and expect exactly that.